tests/unit/command/test_dag.py

```diff
@@ -143,12 +143,12 @@ def test_show_dot(repo):
         "stage;\n"
         "stage;\n"
         "stage;\n"
-        "\"stage: '1'\" -> \"stage: 'a.dvc'\";\n"
-        "\"stage: '2'\" -> \"stage: 'b.dvc'\";\n"
-        "\"stage: '3'\" -> \"stage: 'a.dvc'\";\n"
-        "\"stage: '3'\" -> \"stage: 'b.dvc'\";\n"
-        "\"stage: '4'\" -> \"stage: 'a.dvc'\";\n"
-        "\"stage: '4'\" -> \"stage: '3'\";\n"
+        "\"stage: 'b.dvc'\" -> \"stage: '2'\";\n"
+        "\"stage: 'b.dvc'\" -> \"stage: '3'\";\n"
+        "\"stage: 'a.dvc'\" -> \"stage: '1'\";\n"
+        "\"stage: 'a.dvc'\" -> \"stage: '3'\";\n"
+        "\"stage: 'a.dvc'\" -> \"stage: '4'\";\n"
+        "\"stage: '3'\" -> \"stage: '4'\";\n"
```

Comment:
```suggestion
        "\"stage: '3'\" -> \"stage: '4'\";\n"
        "\"stage: 'b.dvc'\" -> \"stage: '2'\";\n"
        "\"stage: 'b.dvc'\" -> \"stage: '3'\";\n"
        "\"stage: 'a.dvc'\" -> \"stage: '1'\";\n"
        "\"stage: 'a.dvc'\" -> \"stage: '3'\";\n"
        "\"stage: 'a.dvc'\" -> \"stage: '4'\";\n"
```

Reply:
The test suite passes locally for me without changes in the branch. I am not sure if the dot representation is deterministic.

```
pytest -vvv tests/unit/command/test_dag.py
========================== test session starts ==========================
platform linux -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0 -- /home/karthikeyan/stuff/python/dvc/.env/bin/python3.10
cachedir: .pytest_cache
rootdir: /home/karthikeyan/stuff/python/dvc, configfile: pyproject.toml
plugins: lazy-fixture-0.6.3, flaky-3.7.0, forked-1.4.0, cov-3.0.0, mock-3.7.0, anyio-3.5.0, timeout-2.1.0, docker-0.11.0, xdist-2.5.0
collected 15 items                                                      

tests/unit/command/test_dag.py::test_dag[None-_show_ascii] PASSED [  6%]
tests/unit/command/test_dag.py::test_dag[--dot-_show_dot] PASSED  [ 13%]
tests/unit/command/test_dag.py::test_dag[--mermaid-_show_mermaid] PASSED [ 20%]
tests/unit/command/test_dag.py::test_dag[--md-_show_mermaid] PASSED [ 26%]
tests/unit/command/test_dag.py::test_build PASSED                 [ 33%]
tests/unit/command/test_dag.py::test_build_target PASSED          [ 40%]
tests/unit/command/test_dag.py::test_build_target_with_outs PASSED [ 46%]
tests/unit/command/test_dag.py::test_build_granular_target_with_outs PASSED [ 53%]
tests/unit/command/test_dag.py::test_build_full PASSED            [ 60%]
tests/unit/command/test_dag.py::test_build_full_outs[True] PASSED [ 66%]
tests/unit/command/test_dag.py::test_build_full_outs[False] PASSED [ 73%]
tests/unit/command/test_dag.py::test_show_ascii PASSED            [ 80%]
tests/unit/command/test_dag.py::test_show_dot PASSED              [ 86%]
tests/unit/command/test_dag.py::test_show_mermaid PASSED          [ 93%]
tests/unit/command/test_dag.py::test_show_mermaid_markdown PASSED [100%]

========================== 15 passed in 2.94s ===========================
```

Reply:
That's strange. Without the suggested change, It fails on all CI jobs (and my local machine)

Reply:
There are few issues in networkx repo that the graph can be non-deterministic due to hash randomization in Python 3

Reply:
Since the order of the lines doesn't affect the resulting graph visualization, maybe we could just run the assertion with:

```
sorted(_show_dot(repo.index.graph).splitlines()) == sorted(hardcoded_result_lines)
```

Reply:
Thanks @daavoo, I was thinking along similar lines. Added your suggestions with co-author tag.